src/Tests/dotnet-new.Tests/CommonTemplatesTests.cs

```diff
@@ -183,98 +207,93 @@ public void AllCommonProjectsCreate_NoRestore(string expectedTemplateName, strin
 
-            Directory.Delete(workingDir, true);
+            VerificationEngine engine = new VerificationEngine(_logger);
+            await engine.Execute(options).ConfigureAwait(false);
         }
 
         [Fact]
-        public void EditorConfigTests()
+        public async void EditorConfigTests_Empty()
         {
-            string workingDir = CreateTemporaryFolder();
-
-            new DotnetNewCommand(_log, "editorconfig")
-                .WithCustomHive(_fixture.HomeDirectory)
-                .WithWorkingDirectory(workingDir)
-                .Execute()
-                .Should()
-                .ExitWith(0)
-                .And.NotHaveStdErr()
-                .And.HaveStdOut($@"The template ""EditorConfig file"" was created successfully.");
+            TemplateVerifierOptions options = new TemplateVerifierOptions(templateName: "editorconfig")
+            {
+                TemplateSpecificArgs = new[] { "--empty" },
+                SnapshotsDirectory = "Approvals",
+                SettingsDirectory = _fixture.HomeDirectory,
+                VerifyCommandOutput = true,
+            };
 
-            string path = Path.Combine(workingDir, ".editorconfig");
-            string editorConfigContent = File.ReadAllText(path);
-            Assert.Contains("dotnet_naming_rule", editorConfigContent);
-            Assert.Contains("dotnet_style_", editorConfigContent);
-            Assert.Contains("dotnet_naming_symbols", editorConfigContent);
-            File.Delete(path);
+            VerificationEngine engine = new VerificationEngine(_logger);
+            await engine.Execute(options).ConfigureAwait(false);
+        }
 
-            new DotnetNewCommand(_log, "editorconfig", "--empty")
-                .WithCustomHive(_fixture.HomeDirectory)
-                .WithWorkingDirectory(workingDir)
-                .Execute()
-                .Should()
-                .ExitWith(0)
-                .And.NotHaveStdErr()
-                .And.HaveStdOut($@"The template ""EditorConfig file"" was created successfully.");
+        [Fact]
+        public async void EditorConfigTests_Default()
+        {
+            TemplateVerifierOptions options = new TemplateVerifierOptions(templateName: "editorconfig")
+            {
+                SnapshotsDirectory = "Approvals",
+                SettingsDirectory = _fixture.HomeDirectory,
+            }
+            .WithCustomDirectoryVerifier(async (content, contentFetcher) =>
+            {
+                await foreach (var (filePath, scrubbedContent) in contentFetcher.Value)
+                {
+                    filePath.Replace(Path.DirectorySeparatorChar, '/').Should().BeEquivalentTo(@"editorconfig/.editorconfig");
+                    scrubbedContent.Should().Contain("dotnet_naming_rule");
+                    scrubbedContent.Should().Contain("dotnet_style_");
+                    scrubbedContent.Should().Contain("dotnet_naming_symbols");
+                }
+            });
```

Comment:
I think for now the editorconfig content is static. Will the regular test suffice?

Reply:
I added it together with other itemtemplate tests.
I left the original test commented for future reference - it serves as an example of custom callback verifier (and hence a way of testing editorconfig template once/if its content will become dynamic)